### PR TITLE
Rename the generated report file

### PR DESCRIPTION
**Pull request by bigdaz**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -77,5 +77,5 @@ jobs:
             uses: actions/upload-artifact@v3
             with:
                 name: plugin-json
-                path: build/reports/github-dependency-report/github-dependency-manifest.json
+                path: build/reports/github/github-dependency-snapshot.json
```
Comment (sschuberth) [inline .github/workflows/gradle.yml:80]: Mind adding a rationale to the commit message why the new name makes more sense than the old name?

File: LIMITATION.md [hunk 2]
```diff
@@ -1,14 +0,0 @@
-# Current Limitations
-
-These are the current limitations of this project and potential solutions.
-
- - Certain Script Configurations do not work when included builds are present.
-   For example, see the test: `MulitProjectDependencyExtractorTest::project leveraging included builds`
-   The `result.rootComponent.id` is not a `ProjectComponentIdentifier`, so the `projectPath` can not be determined.
-   This could be remedied using a variation of the logic found in the Gradle Enterprise Plugin `ConfigurationResolutionCapturer_5_0`.
```
Comment (sschuberth) [inline LIMITATION.md:8]: Removal of this bit seem unrelated to the commit topic.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (bigdaz) [reply .github/workflows/gradle.yml:80]: The [GitHub Dependency Submission docs](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28) refer to the submitted data as a "snapshot".

> You can submit dependencies in the form of a snapshot. A snapshot is a set of dependencies associated with a commit SHA and other metadata, that reflects the current state of your repository for a commit.

Each snapshot may contain a number of `manifest` entries ([see schema here](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28#create-a-snapshot-of-dependencies-for-a-repository)).

As for changing the directory name, I was trying to avoid redundancy. But perhaps something related some `github-dependency-graph-plugin` would be more informative, given that's the source of the report.
